src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosResourceWrapper.cs

```diff
@@ -105,5 +112,26 @@ internal string GetETagOrVersion()
 
             return base.Version;
         }
+
+        public override void UpdateSearchIndices(IReadOnlyCollection<SearchIndexEntry> searchIndices)
+        {
+            base.UpdateSearchIndices(searchIndices);
+
+            UpdateSortIndex(searchIndices);
+        }
+
+        private void UpdateSortIndex(IReadOnlyCollection<SearchIndexEntry> searchIndices)
+        {
```

Comment:
Given that there is one min and one max per search parameter, I would do a linear scan over the indexes as build up the dictionary as you encounter mins and maxes